src/functTest/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImageFunctionalTest.groovy

```diff
@@ -137,15 +139,14 @@ class DockerBuildImageFunctionalTest extends AbstractGroovyDslFunctionalTest {
         noExceptionThrown()
     }
 
-    @PendingFeature
     def "task can be up-to-date"() {
```

Comment:
We'll need to break up this test method into multiple test cases that describe the specific use cases. At the moment I find it really hard to understand what situation are covered. At the very least I'd expect four different test cases:

- The image has been built. The image file exists. Upon the next execution the task is up to date.
- The image has been built. The image file exists but was deleted manually from the filesystem. Upon the next execution the task is not up to date.
- The image has been built. The image file exists but the image was removed from the registry. Upon the next execution the task is not up to date.
- The image has been built. The image file exists but the value inside of the file was changed manually. Upon the next execution the task is not up to date.

Reply:
Broke into the four test cases you mentioned. A lot of repetition but we need to build twice to get the up-to-date before testing to break it.